tasks/postgresql.yml

```
@@ -26,7 +26,28 @@
 - name: Configure postgresql repository
   package:
     name: "{{ postgresql_yum_repo_url }}"
-  when: ansible_os_family == "RedHat"
+  when: 
+    - ansible_os_family == "RedHat"
+    - ansible_distribution_major_version <= "7"
+
+- block:
+  - name: "Configure postgresql repository key"
+    rpm_key:
+      state: present
+      key:  "{{ postgresql_yum_repo_gpg_key_url }}"
+  
+  - name: Configure postgresql repository
+    package:
+      name: "{{ postgresql_yum_repo_url }}"

```

```suggestion
      name: "{{ postgresql_yum_repo_url }}/{{ postgresql_yum_repo_pkg_name }}"

```

I haven't tried my suggestion. But if I understand your code correctly, there's a package to be installed.

Please be aware that, if upstream provides a newer package but the URL points to an old one, this old one will overwrite the newer. I know, deciding whether you want to package or the repository is a question with no correct answer. But consider looking into the package and just configuring the repository instead of installing the repo package.

---

Your suggestion is fine, but you missed that `postgresql_yum_repo_pkg_name` is already in `postgresql_yum_repo_url` included:
https://github.com/NETWAYS/ansible-role-patroni/pull/4/files#diff-4d1c7ae6092a06448fc8bf930c0e076d4f6ba476aac19b70b3cf6ab66ccd0c3fR3

I'm not sure if I'm getting this correct. I looked into the package and the repositories are in /./etc/yum.repos.d/pgdg-redhat-all.repo defined.

The URL is pointing to https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm to the package pgdg-redhat-repo-latest.noarch.rpm.

The official way to install it is described here: https://www.postgresql.org/download/linux/redhat

Is that correct, or is there a smarter way? @widhalmt

---

It would be possible for EL8 to install PostgreSQL with the standard repositories. We could set a role variable like `patroni_postgresql_install_default` to false and if required install it from default repositories if set to true.

That would need more research and I would implement it in a separate PR.

---

@widhalmt Now, I've got it. Added variable `patroni_postgresql_version_latest` and set it to false. If set to true it'll use the variable `__patroni_postgresql_packages_latest` in vars/<OS-Family>-<Major-Version>.yml.

In that way the user can decide whether to pin down a fixed package version or always install the latest version of the major. 

Also, I established a variable `patroni_install_postgresql_repo` to allow disabling the configuration of the repositories, since they could also be configured by a RedHat Satellite or Foreman.